jenkins/jobs/image_building.pipeline

```diff
@@ -71,47 +72,53 @@ pipeline {
               }
             }
           }
-          stage("Verify the new node image") {
+          stage("Upload the new CI image with -staging suffix") {
             options {
-              timeout(time: 3, unit: 'HOURS')
+              timeout(time: 30, unit: 'MINUTES')
             }
             when {
-              expression { env.IMAGE_TYPE == 'node' }
+              expression { env.IMAGE_TYPE == 'ci' }
             }
             steps {
-              echo "Testing new ${IMAGE_OS} node image"
+              // NOTE: We delete any existing *-staging images before uploading the new one.
```

Comment:
Can we have a situation where we run staging tests from multiple PR and these (try) delete images being used? I don't see us limiting this to 1 at a time?

Reply:
That is a good point. Let me check how to limit it to one, like we do for BML

Reply:
Fixed